awscli/compat.py

```diff
@@ -87,6 +114,8 @@ def compat_open(filename, mode='r', encoding=None):
 
     raw_input = raw_input
 
+    binary_stdin = sys.stdin
```

Comment:
I really don't think it's necessary to add that context manager to `bytes_print`. I tested it out on my windows machine and it appears to have no noticeable effect on the output unless you're trying to write a git command. We could add it, and I don't think it would break things, but it also doesn't help anything.